File: src/main/java/factorio/debugger/DAP/messages/response/DAPSetBreakpointsResponse.java
package factorio.debugger.DAP.messages.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonTypeName;
import factorio.debugger.DAP.messages.DAPAdditionalProperties;
import factorio.debugger.DAP.messages.DAPResponse;
import factorio.debugger.DAP.messages.types.DAPSource;
import factorio.debugger.DAP.messages.types.DAPSourceBreakpoint;

@JsonTypeName("setBreakpoints")
public class DAPSetBreakpointsResponse extends DAPResponse {
    @JsonProperty("body")
    public SetBreakpointsResponse body;
    public static class SetBreakpointsResponse extends DAPAdditionalProperties {
        /**
         * The source location of the breakpoints; either `source.path` or
         * `source.sourceReference` must be specified.
         */
        @JsonProperty("source")
        public DAPSource source;

        /**
         * The code locations of the breakpoints.
         */
        @JsonProperty("breakpoints")
        public DAPSourceBreakpoint[] breakpoints;

        /**
         * Deprecated: The code locations of the breakpoints.
         */
        @JsonProperty("lines")
        public Integer[] lines;

        /**
         * A value of true indicates that the underlying source has been modified
         * which results in new breakpoint locations.
         */
        @JsonProperty("sourceModified")
        public Boolean sourceModified;
    }

    @Override
    public String toString() {
        return super.toString() + ": added "+(body.breakpoints != null ? body.breakpoints.length : 0) + " breakpoints";
    }
}

File: src/main/kotlin/factorio/debugger/DAP/messages/responses/DAPSetBreakpointsResponse.kt
package factorio.debugger.DAP.messages.responses

import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonTypeName
import factorio.debugger.DAP.messages.DAPAdditionalProperties
import factorio.debugger.DAP.messages.types.DAPSource
import factorio.debugger.DAP.messages.types.DAPSourceBreakpoint

@JsonTypeName("setBreakpoints")
class DAPSetBreakpointsResponse : DAPResponse() {
    @JsonProperty("body")
    lateinit var body: SetBreakpointsResponseBody

    class SetBreakpointsResponseBody : DAPAdditionalProperties() {
        /**
         * The source location of the breakpoints; either `source.path` or
         * `source.sourceReference` must be specified.
         */
        @JsonProperty("source")
        var source: DAPSource? = null

        /**
         * The code locations of the breakpoints.
         */
        @JsonProperty("breakpoints")
        lateinit var breakpoints: Array<DAPSourceBreakpoint>

        /**
         * Deprecated: The code locations of the breakpoints.
         */
        @JsonProperty("lines")
        var lines: Array<Int>? = null

        /**
         * A value of true indicates that the underlying source has been modified
         * which results in new breakpoint locations.
         */
        @JsonProperty("sourceModified")
        var sourceModified: Boolean? = null
    }

    override fun toString(): String {
        return "${super.toString()}: added ${body.breakpoints.size} breakpoints"
    }
}
